### PR TITLE
Remove PlatformLayout

### PR DESCRIPTION
**Pull request by minborg**
This PR suggests removing the class `PlatformLayout`.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Change must be properly reviewed (1 review required, with at least 1 [Committer](https://openjdk.org/bylaws#committer))

### Reviewers
 * [Maurizio Cimadamore](https://openjdk.org/census#mcimadamore) (@mcimadamore - Committer)
 * [Jorn Vernee](https://openjdk.org/census#jvernee) (@JornVernee - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/panama-foreign pull/794/head:pull/794` \
`$ git checkout pull/794`

Update a local copy of the PR: \
`$ git checkout pull/794` \
`$ git pull https://git.openjdk.org/panama-foreign pull/794/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 794`

View PR using the GUI difftool: \
`$ git pr show -t 794`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/panama-foreign/pull/794.diff">https://git.openjdk.org/panama-foreign/pull/794.diff</a>

</details>

### CODE REVIEW

**Comment by JornVernee** [comment]
I've been thinking for a while now that `PlatformLayout` should just be removed: https://bugs.openjdk.org/browse/JDK-8291499 It is a remnant of a time where we exposed layout constant for each platform.

In the implementation we only use one or two layouts for each platform. I think `PlatformLayout` should just be removed, and the needed layouts moved to the particular `CallArranger` class where they are needed. So, if you want to work on `PlatformLayouts` I suggest removing it instead :)

Last time I looked into this, the tricky part was that there was a test that used these layouts as test values. That test will have to do something else.

File: src/java.base/share/classes/jdk/internal/foreign/PlatformLayouts.java [hunk 1]
```diff
@@ -233,42 +215,44 @@ private RISCV64() {
...
-                .withTargetLayout(MemoryLayout.sequenceLayout(C_CHAR));
+        public static final ValueLayout.OfAddress C_POINTER = cPointer(C_CHAR);
 
-        /**
-         * The {@code va_list} native type, as it is passed to a function.
-         */
-        public static final ValueLayout.OfAddress C_VA_LIST = RISCV64.C_POINTER;
     }
+
+    private static ValueLayout.OfAddress cPointer(ValueLayout targetLayoutElementLayout) {
```
Comment (minborg) [inline src/java.base/share/classes/jdk/internal/foreign/PlatformLayouts.java:252]: Maybe there is a better name? `elementLayout` perhaps?

Comment (mcimadamore) [reply src/java.base/share/classes/jdk/internal/foreign/PlatformLayouts.java:252]: I think you can just omit the parameter here - and use JAVA_BYTE. We're not interested in what the layout is - the main thing is that it's unbounded (and that it has minimum alignment). Even using a padding layout of Long.MAX_VALUE would work :-)

File: test/jdk/java/foreign/callarranger/platform/PlatformLayouts.java [hunk 2]
```diff
@@ -55,53 +54,47 @@ private SysV() {
...
          */
-        public static final ValueLayout.OfDouble C_DOUBLE = ValueLayout.JAVA_DOUBLE.withBitAlignment(64);
+        public static final ValueLayout.OfDouble C_DOUBLE = ValueLayout.JAVA_DOUBLE;
 
         /**
          * The {@code T*} native type.
          */
-        public static final ValueLayout.OfAddress C_POINTER = ValueLayout.ADDRESS.withBitAlignment(64)
-                .withTargetLayout(MemoryLayout.sequenceLayout(C_CHAR));
+        public static final ValueLayout.OfAddress C_POINTER = cPointer(C_CHAR);
```
Comment (mcimadamore) [inline test/jdk/java/foreign/callarranger/platform/PlatformLayouts.java:87]: Can you use SharedUtils.C_POINTER here? It's the same thing.

**Review by mcimadamore — APPROVED** [review]

**Review by JornVernee — APPROVED** [review]

**Comment by mcimadamore** [comment]
Btw, why is this PR called "Remove PlatformLayout" given that it does not remove it?

**Pull request closed without merging**